tests/Unit/FeedXMLGenerationTest.php

```diff
@@ -0,0 +1,405 @@
+		$product->save();
+
+		$xml = $image_link_method( $product );
+		$this->assertEquals( "<g:image_link><![CDATA[http://example.org/wp-content/uploads/product_image.png]]></g:image_link>", $xml );
+	}
+
+	/**
+	 * @group feed
+	 */
+	public function testPropertyAvailabiltiyXML() {
+		$availability_method = $this->getProductsXmlFeedAttributeMethod( 'g:availability' );
+		$product             = \WC_Helper_Product::create_simple_product();
+
+		// Set different statuses and test.
+		$product->set_stock_status( 'instock' );
+		$xml = $availability_method( $product );
+		$this->assertEquals( "<g:availability>in stock</g:availability>", $xml );
+
+		$product->set_stock_status( 'outofstock' );
+		$xml = $availability_method( $product );
+		// create_simple_product gives the product 'Uncategorized' type.
+		$this->assertEquals( "<g:availability>out of stock</g:availability>", $xml );
+
+		$product->set_stock_status( 'onbackorder' );
+		$xml = $availability_method( $product );
+		// create_simple_product gives the product 'Uncategorized' type.
+		$this->assertEquals( "<g:availability>preorder</g:availability>", $xml );
+	}
+
+	/**
+	 * @group feed
+	 */
+	public function testPropertyPriceXML() {
+		$price_method = $this->getProductsXmlFeedAttributeMethod( 'g:price' );
+		$product      = \WC_Helper_Product::create_simple_product( true, array( "regular_price" => 15 ) );
+		$xml          = $price_method( $product );
+		$this->assertEquals( "<g:price>15USD</g:price>", $xml );
+	}
+
+	/**
+	 * @group feed
+	 */
+	public function testPropertySalePriceXML() {
+		$sale_price_method = $this->getProductsXmlFeedAttributeMethod( 'sale_price' );
+
+		// No sale price is set.
+		$product = \WC_Helper_Product::create_simple_product( true, array( "regular_price" => 15 ) );
+		$xml     = $sale_price_method( $product );
+		$this->assertEquals( "", $xml );
+
+		// Sale price set.
+		$product = \WC_Helper_Product::create_simple_product(
+			true,
+			array(
+				"regular_price" => 15,
+				"sale_price"    => 5
+			)
+		);
+		$xml     = $sale_price_method( $product );
+		$this->assertEquals( "<sale_price>5USD</sale_price>", $xml );
+	}
+
+	/**
+	 * @group feed
+	 */
+	public function testPropertySalePriceVariableProductXML() {
+		$price_method      = $this->getProductsXmlFeedAttributeMethod( 'g:price' );
+		$product           = new \WC_Product_Variable();
+		$variation_product = \WC_Helper_Product::create_variation_product( $product );
+		/* In UT flow we need to fetch the product again from the DB after creation.
+		 * This ensures correct initialization of visible variations.
+		 * Without that the variable price methods think that we don't have visible children.
+		 * Quirk of create_variation_product.
+		 */
+		$product = wc_get_product( $variation_product->get_id() );
+		$xml     = $price_method( $product );
+		// 10.00USD is the cheapest variation created by create_variation_product
+		$this->assertEquals( "<g:price>10.00USD</g:price>", $xml );
```

Comment:
For a simple product we expect the price to be shown without decimals `15USD` whereas for a variable product we are testing it with decimals `10.00USD`. Shouldn't those both be in the same format in the feed?

Reply:
Good catch. Prices in case of `get_variation_regular_price` go through `wc_format_decimal`. I have created an issue to review and tackle this: https://github.com/woocommerce/pinterest-for-woocommerce/issues/325